server/src/test/java/io/spine/examples/shareaware/server/investment/SharesPurchaseTest.java

```diff
@@ -0,0 +1,305 @@
+import io.spine.examples.shareaware.wallet.Wallet;
+import io.spine.examples.shareaware.wallet.command.CancelMoneyReservation;
+import io.spine.examples.shareaware.wallet.command.DebitReservedMoney;
+import io.spine.examples.shareaware.wallet.command.ReserveMoney;
+import io.spine.examples.shareaware.wallet.event.MoneyReservationCanceled;
+import io.spine.examples.shareaware.wallet.event.MoneyReserved;
+import io.spine.examples.shareaware.wallet.event.ReservedMoneyDebited;
+import io.spine.examples.shareaware.wallet.rejection.Rejections.InsufficientFunds;
+import io.spine.server.BoundedContextBuilder;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import io.spine.testing.server.model.ModelTests;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.shareaware.server.investment.given.InvestmentTestEnv.*;
+import static io.spine.examples.shareaware.server.given.WalletTestEnv.setUpReplenishedWallet;
+import static io.spine.examples.shareaware.server.given.WalletTestEnv.setUpWallet;
+
+@DisplayName("`SharesPurchase` should")
+public final class SharesPurchaseTest extends ContextAwareTest {
+
+    @BeforeAll
+    static void beforeAll() {
+        ModelTests.dropAllModels();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        ModelTests.dropAllModels();
+    }
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return PurchaseTestContext.newBuilder();
+    }
+
+    @Nested
+    @DisplayName("reduce the wallet balance")
+    class WithdrawMoneyFromBalance {
+
+        @Test
+        @DisplayName("for the purchase price amount")
+        void event() {
+            Wallet wallet = setUpReplenishedWallet(context());
+            UserId user = wallet.getId()
+                                .getOwner();
+            PurchaseShares firstPurchase = purchaseSharesFor(user);
+            PurchaseShares secondPurchase = purchaseSharesFor(user);
+            Wallet walletAfterPurchase =
+                    walletAfter(firstPurchase, secondPurchase, wallet);
+            context().receivesCommands(firstPurchase, secondPurchase);
+
+            context().assertState(wallet.getId(), walletAfterPurchase);
+        }
+
+        @Test
+        @DisplayName("emitting the `MoneyReserved` event")
+        void moneyReserved() {
+            Wallet wallet = setUpReplenishedWallet(context());
+            PurchaseShares command = purchaseSharesFor(wallet.getId()
+                                                             .getOwner());
+            MoneyReserved expected = moneyReservedBy(command);
+            context().receivesCommand(command);
+
+            context().assertEvent(expected);
+        }
+
+        @Test
+        @DisplayName("emitting the `ReservedMoneyDebited` event")
+        void debitMoney() {
```

Comment:
Based on other method names:
```suggestion
        void reservedMoneyDebited() {
```
or
```suggestion
        void moneyDebited() {
```